flask_babel/__init__.py

```diff
@@ -179,21 +186,43 @@ def default_timezone(self):
         """
         return timezone(self.app.config['BABEL_DEFAULT_TIMEZONE'])
 
+    @property
+    def translation_directories(self):
+        directories = self.app.config.get(
+            'BABEL_TRANSLATION_DIRECTORIES',
+            'translations'
+        ).split(';')
+
+        for path in directories:
+            if os.path.isabs(path):
+                yield path
+            else:
+                yield os.path.join(self.app.root_path, path)
+
 
 def get_translations():
     """Returns the correct gettext translations that should be used for
     this request.  This will never fail and return a dummy translation
     object if used outside of the request or if a translation cannot be
     found.
     """
-    ctx = _request_ctx_stack.top
-    if ctx is None:
-        return None
+    ctx = _get_current_context()
+
     translations = getattr(ctx, 'babel_translations', None)
     if translations is None:
-        dirname = os.path.join(ctx.app.root_path, 'translations')
-        translations = support.Translations.load(dirname, [get_locale()])
+        translations = support.Translations()
+
+        babel = current_app.extensions['babel']
```

Comment:
The docstring states `This will never fail and return a dummy translation object if used outside of the request or if a translation cannot be found.`, however the attribute access on `current_app` breaks this behavior (and my apps).